### ISSUE TITLE
Core dumps on ubuntu on Rpi4

### ISSUE THREAD

**Issue by manicmoose** [unpoller/unpoller#385]
The current version v2.1.3 rev: 884 runs fine on a Rpi4 on 32-bit Raspberry Pi OS.
Migrating the same install (with same config file) over to another Rpi4 running 64-bit Ubuntu - it core dumps on start:

```
# uname -a
Linux pi4.my.lan 5.4.0-1047-raspi #52-Ubuntu SMP PREEMPT Wed Nov 24 08:16:38 UTC 2021 aarch64 aarch64 aarch64 GNU/Linux

# dpkg -l unifi-poller
||/ Name           Version      Architecture Description
+++-==============-============-============-====================================================================
ii  unifi-poller   2.1.3-884    arm64        Polls a UniFi controller, exports metrics to InfluxDB and Prometheus


# strace /usr/bin/unifi-poller
execve("/usr/bin/unifi-poller", ["/usr/bin/unifi-poller"], 0xffffd72a92b0 /* 38 vars */) = 0
mmap(0xb10000, 3774884, PROT_READ|PROT_WRITE|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0xb10000
--- SIGILL {si_signo=SIGILL, si_code=ILL_ILLOPC, si_addr=0xea8214} ---
+++ killed by SIGILL (core dumped) +++
Illegal instruction (core dumped)

```

**Comment by davidnewhall** [comment]
Can you elaborate on _migrating_? Does it work if you install it from a package?

**Comment by davidnewhall** [comment]
oh it's SIGILL. This is a weird issue that comes up on some arm64 systems. Running the 32 bit version (armhf) will work.

**Comment by manicmoose** [comment]
Yeah, I was already running the packaged version.
Tried the 32bit one as you suggested and it worked.  Thanks for that.

For those with a similar issue before you can install it you need to run:
`dpkg --add-architecture armhf
`

Also, I think there's a bug - if you explicitely state "debug = false" in the config file, it stil outputs lots of info to syslog.
If you comment out all 'debug' statements, then the default works and all is nice and quiet.
So, I think if "debug" is found in the config file at all, it defaults to debug=true.

**Comment by davidnewhall** [comment]
If debug is set true in the config or environment it will enable the extra output. Thanks for sharing your fix!

**Comment by manicmoose** [comment]
The bug is:
 - if debug is set *at all* (to either 'true' **or** 'false') in the config file, it defaults to 'on'.
You have to comment it out to set it to 'off'.

**Comment by davidnewhall** [comment]
I'm not able to reproduce that behavior.

**Comment by manicmoose** [comment]
Yeah, after a reboot I'm not able to reproduce it anymore either.  Odd...

**Comment by davidnewhall** [comment]
I think I found another fix/workaround for the SIGILL on arm64 platforms.

```
sudo apt install upx || sudo yum install upx
sudo upx -d /usr/bin/unifi-poller
```

**Comment by manicmoose** [comment]
Nice one, David - it works!!
I guess I'll have to remember to uncompress the binary again after each upgrade.. ;-)

FYI: on an Ubuntu system the package to install is "upx-ucl"

Thanks again!
